code/android-core-library/src/main/java/com/adobe/marketing/mobile/Log.java

```diff
@@ -57,77 +57,102 @@ static LoggingMode getLogLevel() {
      * Used to print more verbose information. Info logging is expected to follow end-to-end every method an event hits.
      * Prints information to the console only when the SDK is in LoggingMode: VERBOSE
      *
-     * @param source the source of the information to be logged
-     * @param format the string format to be logged
-     * @param params values to be inserted into the format
+     * @param extension the extension name
+     * @param source    the source of the information to be logged
+     * @param format    the string format to be logged
+     * @param params    values to be inserted into the format
      * @see LoggingMode
      */
-    public static void trace(final String source, final String format, final Object... params) {
+    public static void trace(final String extension, final String source, final String format, final Object... params) {
         if (loggingService != null && loggingMode.id >= LoggingMode.VERBOSE.id) {
             try {
-                loggingService.trace(source, String.format(format, params));
+                loggingService.trace(extension + "-" + source, String.format(format, params));
             } catch (Exception e) {
                 loggingService.trace(source, format);
             }
         }
     }
 
+    @Deprecated
+    public static void trace(final String source, final String format, final Object... params) {
+        trace(null, source, format, params);
+    }
+
+
     /**
      * Information provided to the debug method should contain high-level details about the data being processed.
      * Prints information to the console only when the SDK is in LoggingMode: VERBOSE, DEBUG
      *
-     * @param source the source of the information to be logged
-     * @param format the string format to be logged
-     * @param params values to be inserted into the format
+     * @param extension the extension name
+     * @param source    the source of the information to be logged
+     * @param format    the string format to be logged
+     * @param params    values to be inserted into the format
      * @see LoggingMode
      */
-    public static void debug(final String source, final String format, final Object... params) {
+    public static void debug(final String extension, final String source, final String format, final Object... params) {
         if (loggingService != null && loggingMode.id >= LoggingMode.DEBUG.id) {
             try {
-                loggingService.debug(source, String.format(format, params));
+                loggingService.debug(extension + "-" + source, String.format(format, params));
```

Comment:
Can you do a null check for extension to avoid printing `null`?

Reply:
Ops, it will print the `null` String. Ok, how about using an empty String instead?